jupyter_dashboards/nbextension/notebook/dashboard-view/dashboard-metadata.js

```diff
@@ -0,0 +1,235 @@
+            var node = $(el).data('_gridstack_node');
+            return {
+                col: node.x,
+                row: node.y,
+                width: node.width,
+                height: node.height
+            };
+        });
+
+        $('.cell.grid-stack-item').each(function(idx) {
+            var layout = $.extend({}, widgetData[idx]); // clone `data` object
+            _updateCellMetadata($(this), layout);
+        });
+
+        IPython.notebook.set_dirty(true);
+    }
+
+    function _setDashboardLayout(dbLayout) {
+        if (_validValue(DASHBOARD_LAYOUT, dbLayout)) {
+            IPython.notebook.metadata.urth.dashboard.layout = dbLayout;
+        } else {
+            throw new Error('Invalid dashboard layout:', dbLayout);
+        }
+    }
+
+    function _setLayoutStrategy(strategy) {
+        if (_validValue(LAYOUT_STRATEGY, strategy)) {
+            IPython.notebook.metadata.urth.dashboard.layoutStrategy = strategy;
+        } else {
+            throw new Error('Invalid layout strategy:', strategy);
+        }
+    }
+
+    function _showCell(cells) {
+        $(cells).each(function() {
+            var metadata = _getCellMetadata($(this)).urth.dashboard;
+            metadata.layout = {};
+            delete metadata.hidden;
+        });
+    }
+
+    function _updateCellMetadata($cells, layout) {
+        if (arguments.length === 1) {
+            layout = $cells; // first argument is optional
+            $cells = $('.cell');
+        } else {
+            $cells = $($cells); // force to jquery
+        }
+
+        $cells.each(function(i, cell) {
+            var metadata = _getCellMetadata($(cell)).urth.dashboard;
+
+            // only update the layout if not hidden
+            if (layout && !metadata.hidden) {
+                metadata.layout = metadata.layout || {};
+                CELL_PROPERTIES.forEach(function(prop) {
+                    if (layout.hasOwnProperty(prop)) {
+                        metadata.layout[prop] = layout[prop];
+                    }
+                });
+            } else {
+                delete metadata.layout;
+                metadata.hidden = true;
+            }
+        });
+        IPython.notebook.set_dirty(true);
+    }
+
+    function _validValue(obj, value) {
+        return Object.keys(obj).map(function(key) {
+                return obj[key];
+            }).indexOf(value) !== -1;
+    }
+
+    return {
+        DASHBOARD_LAYOUT: DASHBOARD_LAYOUT,
+        LAYOUT_STRATEGY: LAYOUT_STRATEGY,
+
+        /**
+         * @param  {jQuery} $cell - notebook cell or element inside a notebook cell
+         * @return {Object} layout positioning for the specified cell
+         */
+        getCellLayout: function($cell) {
+            return _getCellMetadata($cell).urth.dashboard.layout;
+        },
+        /**
+         * @return {string} dashboard layout type (defaults to grid layout if not set in notebook metadata)
+         */
+        getDashboardLayout: function() {
```

Comment:
We should start using the ES5 getter/setter syntax for these types of "properties" (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get).  Allows consumers to write code like `foo = metadata.dashboardLayout` to get and `metadata.dashboardLayout = foo` to set.

Also, getters and setters for same properties should be defined next to each other in this file.